src/functional/java/reposense/Entry.java

```diff
@@ -21,63 +25,71 @@
     private static final String EXPECTED_FOLDER = "expected";
 
     @Test
-    public void test() {
+    public void test() throws Exception {
         generateReport();
         String actualRelativeDir = getRelativeDir();
-        File actualFiles = new File(getClass().getClassLoader().getResource("expected").getFile());
+        Path actualFiles = Paths.get(getClass().getClassLoader().getResource("expected").toURI());
         verifyAllJson(actualFiles, actualRelativeDir);
         FileUtil.deleteDirectory(FT_TEMP_DIR);
     }
 
-    private void generateReport() {
-        File configFile = new File(getClass().getClassLoader().getResource("sample_full.csv").getFile());
+    private void generateReport() throws Exception {
+        Path configFilePath = Paths.get(getClass().getClassLoader().getResource("sample_full.csv").toURI());
         Calendar c = Calendar.getInstance();
         c.set(2017, 6, 1);
         Date fromDate = c.getTime();
         c.set(2017, 10, 30);
         Date toDate = c.getTime();
-        List<RepoConfiguration> configs = CsvConfigurationBuilder.buildConfigs(configFile, fromDate, toDate);
+        List<RepoConfiguration> configs = CsvConfigurationBuilder.buildConfigs(configFilePath, fromDate, toDate);
         RepoInfoFileGenerator.generateReposReport(configs, FT_TEMP_DIR);
     }
 
-    private void verifyAllJson(File expectedDirectory, String actualRelative) {
-        for (File file : expectedDirectory.listFiles()) {
-            if (file.isDirectory()) {
-                verifyAllJson(file, actualRelative);
-            } else {
-                if (!file.getName().endsWith(".js")) {
-                    continue;
+    private void verifyAllJson(Path expectedDirectory, String actualRelative) {
+        try (Stream<Path> pathStream = Files.list(expectedDirectory)) {
+            pathStream.forEach(filePath -> {
```

Comment:
Wondering if we should remain using a standard `for loop` instead (see [8.1. Avoid Blocks of Code in Lambda’s Body](http://www.baeldung.com/java-8-lambda-expressions-tips). :P

Reply:
@Zhiyuan-Amos interested in helping out with RepoSense? ^^

Good catch, I agree that it defeats the purpose of lambda expression.

Reply:
@yamidark 

He mean retain the old for loop instead of using lambda expression e.g.
```
for (...) {
  // that whole chunk...
}
```
AFAIK lambda expression doesn't speed up your code and is often
So throwing the chunk of code in that lambda's body defeat the purpose of functional programming and also doesn't help much in readability.